### PR TITLE
Added BST Sort in C-Plus-Plus

### LINKED ISSUE #2570: Binary search tree(No. Of leaf nodes and sorting)
## 🚀 Feature
I'm a GSSoC'21 participant.I can implement the following operations on bst in C and C++

1. Insert into bst
2. Count the number of leaf nodes
3. Sort a list of numbers
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#reporting-new-issues)?

Yes

## Motivation

Binary searc tree is one of the important data structures

## Pitch

This special type of binary tree is really important for solving various problems

### PR DESCRIPTION
**Pull request by aish2002**
Fixes #2570

### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

Sorts a list of numbers using binary search tree

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

#2570

### CODE REVIEW

**Comment by aish2002** [comment]
@HarshCasper , @bhav09 , @siddharth25pandey  , Kindly review the PR

File: C-Plus-Plus/sort/BST_Sort.cpp [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+    }
+
+    printf("\nSorted list : ");
+    inorder(root);
+    printf("\n");
+
+    return 0;
+}
+
+/*
```
Comment (siddharth25pandey) [inline C-Plus-Plus/sort/BST_Sort.cpp]: Remove the blank lines

File: C-Plus-Plus/sort/BST_Sort.cpp [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+    Sample input:
+    7
+    1 2 5 3 7 4 6
+
+    Sample output:
+    1 2 3 4 5 6 7
+*/
+
+/*
```
Comment (siddharth25pandey) [inline C-Plus-Plus/sort/BST_Sort.cpp]: Remove the blank lines

**Review by siddharth25pandey — CHANGES_REQUESTED** [review]
@aish2002 Remove the unnecessary blank lines
Rest is fine :)

**Comment by aish2002** [comment]
I have removed the unnecessary blank lines. @siddharth25pandey  kindly re-review the PR

**Review by siddharth25pandey — APPROVED** [review]
@aish2002 LGTM, able to merge

**Comment by siddharth25pandey** [comment]
@all-contributors please add @aish2002 for her code.

**Pull request merged**